rotkehlchen/globaldb/handler.py

```diff
@@ -1634,54 +1636,56 @@ def soft_reset_assets_list() -> tuple[bool, str]:
         builtin_database = root_dir / 'data' / 'global.db'
         detach_database = 'DETACH DATABASE "clean_db";'
 
-        try:
-            with GlobalDBHandler().conn.read_ctx() as read_cursor:
-                read_cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
-                # Check that versions match
-                query = read_cursor.execute('SELECT value from clean_db.settings WHERE name="version";')  # noqa: E501
-                version = query.fetchone()
-                if version is None or int(version[0]) != globaldb_get_setting_value(read_cursor, 'version', GLOBAL_DB_VERSION):  # noqa: E501
+        with GlobalDBHandler().packaged_db_lock:
+            try:
+                with GlobalDBHandler().conn.read_ctx() as read_cursor:
+                    read_cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
+                    # Check that versions match
+                    query = read_cursor.execute('SELECT value from clean_db.settings WHERE name="version";')  # noqa: E501
+                    version = query.fetchone()
+                    if version is None or int(version[0]) != globaldb_get_setting_value(read_cursor, 'version', GLOBAL_DB_VERSION):  # noqa: E501
+                        read_cursor.execute(detach_database)
```

Comment:
Do the same thing you did above

1. Delete the `detach_database` variable as it's just used in 1 place.
2. Move it to the finally and just use the literal string.

```suggestion
```